### PR TITLE
DEC-1352: Add confirmation modal on flagging account for moderation

### PR DESCRIPTION
**Pull request by shelegdmitriy**
Closes:
- https://pagodaplatform.atlassian.net/browse/DEC-1352;
- #315

### CODE REVIEW

**Comment by calebjacob** [comment]
@shelegdmitriy Let's make use of the Radix Dialog primitive for the modal: https://www.radix-ui.com/docs/primitives/components/dialog

**Comment by shelegdmitriy** [comment]
@calebjacob indeed! I left this is this state to focus on main functional part. As soon as we'll get a design for DIG.Modal I'll fix this

**Comment by charleslavon** [comment]
hey @shelegdmitriy To make review/merging easier,  use the ability to save/create a PR as `Draft` as oppose to writing Don't Merge in the description.

File: src/Flagged/Modal.jsx
Comment (gabehamilton) [inline src/Flagged/Modal.jsx:49]: Can you switch to the new structure described [here](https://github.com/near/near-discovery/discussions/386)?

```suggestion
         [filterFlaggedAccounts]: ""
```

Comment (shelegdmitriy) [reply src/Flagged/Modal.jsx:49]: And it will mean that we want ti hide all the posts without reporting, right @gabehamilton ?

Comment (gabehamilton) [reply src/Flagged/Modal.jsx]: Yes, that's correct.

File: src/AccountProfile.jsx
Comment (gabehamilton) [inline src/AccountProfile.jsx:4]: maybe this prop should be named something like `showActionButtons`?

Comment (shelegdmitriy) [reply src/AccountProfile.jsx:4]: This prop defines the place where FlagButton is clicked. We need to be sure that flag account button is clicked on ActivityFeed

Comment (gabehamilton) [reply src/AccountProfile.jsx:4]: Right. I was thinking we may show the AccountProfile lots of places. With some we'll want to show those buttons (like on the ActivityFeed), others we won't.

Comment (shelegdmitriy) [reply src/AccountProfile.jsx:4]: Make sense

**Comment by gabehamilton** [comment]
I think the Confirm and Cancel buttons need more functionality. Both lead to the save data modal and they need to close the 'Do you want to hide all posts of this account?' modal so you can interact with the save data modal.

**Comment by gabehamilton** [comment]
Looks really good. Normally I'd say it's ready to merge. However, I just re-opened the data discussion https://github.com/near/near-discovery/discussions/386
Let's wait to merge this PR until we can resolve that discussion and store the hide account data in the final format.

**Comment by shelegdmitriy** [comment]
After a small discussion with @gabehamilton I decided to add a couple if things here:

1. "Not Now" button or "X" button to simply close this Dialog without reporting.
2. Add blur effect on background. This change makes this Dialog to looks exactly like NotificationAlert (I've added blur effect by default but this can be controlled by developer).

All this things could be added in just 1 line of code here after #357 will be merged.

cc @charleslavon

File: src/Flagged/Modal.jsx
Comment (gabehamilton) [inline src/Flagged/Modal.jsx:49]: This data won't go in the index. You can take out this line (and the matching brace.

File: src/Flagged/Modal.jsx
Comment (gabehamilton) [inline src/Flagged/Modal.jsx:14]: You don't need the existing accounts (or posts). Each user's self-moderation data is going into the moderate object with the accounts as keys. The `set` method merges keys into that object when it stores it in the contract.

File: src/Flagged/Modal.jsx
Comment (gabehamilton) [inline src/Flagged/Modal.jsx:51]: Let's make this `report` for now. It can get more options when we do the reporting flow. In that flow `report` will be the value to store for a generic report (without choosing a reason).
```suggestion
        [filterFlaggedAccounts]: "report",
```

File: src/Flagged/Modal.jsx
Comment (shelegdmitriy) [reply src/Flagged/Modal.jsx:49]: But why? We've been making a `Social.set()` all the time. @gabehamilton could you please provide more info here?

Comment (gabehamilton) [reply src/Flagged/Modal.jsx:29]: Latest update. Let's use this format for compatibility with gateways that may want to use Social.index in their implementation.
```
    "index": {
      "moderation": `{\"key\":\"reported\",\"value\":{\"path\":\"${reportedAccountId}\", \"label\": \"report\"}}`
    },
    "moderation": {
      [reportedAccountId]: "report"
    },
```

**Review by gabehamilton — APPROVED** [review]

File: src/Flagged/Modal.jsx
Comment (shelegdmitriy) [inline src/Flagged/Modal.jsx:33]:
```suggestion
    index: {
      moderation: {
        key: "reported",
        value: {
          path: reportedAccountId,
          label: "report"
        },
        [reportedAccountId]: "report",
      },
    },
    moderation: {
      [reportedAccountId]: "report",
    },
```


@gabehamilton according to your comment below is this what you meant?

Comment (gabehamilton) [reply src/Flagged/Modal.jsx:33]: The index.moderation value needs to be a string for the index standard.
```suggestion
      index: {
          moderation: `{"key":"reported","value":{"path":"${reportedAccountId}", "label": "report"}}`
      },
      moderation: {
          [reportedAccountId]: "report",
      },

```

File: src/AccountProfile.jsx
Comment (roshaans) [reply src/AccountProfile.jsx]: ll

File: src/AccountProfile.jsx
Comment (roshaans) [inline src/AccountProfile.jsx:94]: creating a thread with gh.nvim

File: src/Flagged/Modal.jsx
Comment (shelegdmitriy) [reply src/Flagged/Modal.jsx:33]: Thanks @gabehamilton!

**Pull request merged**
